Add Custom Consul services definition

kamedodji opened the pull request:
Consul services definition is now hash of hashes,
instead of array of hashes. This way, service can be
easily activated or disabled right now...

Commit:
Add Custom Consul services definition

Consul services definition is now hash of hashes,
instead of array of hashes. This way, service can be
easily activated or disabled right now...

rezib commented on src/inst/opts/consul.asc:
```diff
@@ -0,0 +1,52 @@
+=== Custom Consul services
+
+Consul services check is based on defintion as per file `puppet-hpc/hieradata/common.yaml` extract:
```
1/ There is a typo on word definition
2/ Do we really want to maintain a duplicate extract of puppet-hpc here? Just one fake example on custom additional service definition is fine.

kamedodji replied:
fixed

rezib commented on src/inst/opts/consul.asc:
```diff
@@ -0,0 +1,52 @@
+----
+
+Any of these service can be disabled with empty hash definition.
+For example, to disable web-system service, just add to `$ADMIN/hpc-privatedata/hieradata/$CLUSTER/cluster.yaml`:
```
Please wrap and add empty line.

kamedodji replied:
new commit!

rezib commented on src/inst/gen_serv_nodes/consul.asc:
```diff
@@ -97,3 +97,5 @@ Finally, check DNS requests on virtual zone are managed by Consul with:
 The output must report multiple generic service nodes static IP addresses in
 random order.
+
+To advanced custom consul services definition, please <<_custom_consul_services,go to this section...>>
```
I really prefer unaliased internal reference (ie. `<<_custom_consul_services>>`) since we build the documents with `:xrefstyle: full` which produces links that can also be read when the document is printed.

Commit:
use unaliased internal reference

indeed, since we build documents with :xrefstyle: full, it's
better to use unaliased internal reference of
<<_custom_consul_services>>

Commit:
add empty line

Commit:
typo fixed and no more duplicate subservices hiera

rezib commented on src/inst/opts/consul.asc:
```diff
@@ -1,38 +1,16 @@
 
-Consul services check is based on defintion as per file `puppet-hpc/hieradata/common.yaml` extract:
+Consul services check is based on hash of hases hiera named : `profiles::consul::server::subservices` and defined
+ in file `$ADMIN/puppet-hpc/hieradata/common.yaml`:
```
1/ You introduced a new typo with hases
2/ The sentence with 2 columns (`:`) is weird
3/ The wrapping is not correct
4/ You don't explain the purpose of the following yaml excerpt

rezib commented on src/inst/opts/consul.asc:
```diff
@@ -44,6 +44,7 @@ profiles::consul::server::subservices:
 
 Any of these service can be disabled with empty hash definition.
 For example, to disable web-system service, just add to `$ADMIN/hpc-privatedata/hieradata/$CLUSTER/cluster.yaml`:
+
```
You missed the wrapping here.

Commit:
fix typo and missed wrapping

rezib commented on src/inst/opts/consul.asc:
```diff
@@ -1,7 +1,7 @@
-Consul services check is based on hash of hases hiera named : `profiles::consul::server::subservices` and defined
- in file `$ADMIN/puppet-hpc/hieradata/common.yaml`:
+Consul services check is based on hash of hashes hiera named `profiles::consul::server::subservices`
+and defined in file `$ADMIN/puppet-hpc/hieradata/common.yaml`:
```
The sentence ends with a column, hence readers believe the following yaml excerpt is the content of this file, but it is not the case.

kamedodji replied:
Pls check latest update...

rezib commented on src/inst/opts/consul.asc:
```diff
@@ -20,6 +20,8 @@ profiles::consul::server::subservices:
       timeout: '1s'
 ----
 
+This defined two services, `http` and `web-system`, with there associated check parameters.
```
I find it weird to get the purpose of the yaml code _after_ it. I think it is more clear if this sentence is _before_ the yaml code.

kamedodji replied:
Pls check latest update...

Commit:
fixed again typo and wrapping!

rezib commented on src/inst/opts/consul.asc:
```diff
@@ -0,0 +1,37 @@
+`profiles::consul::server::subservices` and defined in file 
+`$ADMIN/puppet-hpc/hieradata/common.yaml`.
+
+For instance find bellow excerpt of two services definition, 
```
There are 2 typos:
1/ s/bellow/below/
2/ s/definition/definitions/

kamedodji replied:
fixed!

Commit:
fixed again latest typo!
